### PR TITLE
fix: incorrect JSDoc for `Deno.noColor`

### LINKED ISSUE #10360: Deno.noColor is constant.
`Deno.noColor` doesn't change when `NO_COLOR` environment variable changes. I think it should because the environment variables can be modified at runtime and `Deno.noColor` should adapt when `NO_COLOR` changes. Should be a getter.

```ts
console.log(Deno.noColor);
Deno.env.set("NO_COLOR", "1");
console.log(Deno.noColor);
```

Prints
```ts
false
false
```

Runned with

```sh
deno run --no-check --allow-env 'data:application/javascript;utf8,console.log(Deno.noColor);Deno.env.set("NO_COLOR","1");console.log(Deno.noColor);'
```

Allowing colors to be disabled at runtime is a useful feature, and in my opinion should be possible.

### PR DESCRIPTION
**Pull request by sno2**
fixes #10360

### CODE REVIEW

**Review by kitsonk — CHANGES_REQUESTED** [review]
The unrelated formatting changes should be removed from the PR.

File: cli/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -89,7 +89,8 @@ declare namespace Deno {
   /** The current process id of the runtime. */
   export const pid: number;
 
-  /** Reflects the `NO_COLOR` environment variable.
+  /** If the ENV contained `NO_COLOR` with any value at the initial
+   * initial runtime, then set to true. Otherwise, this is false.
```
Comment (lucacasonato) [inline cli/dts/lib.deno.ns.d.ts:93]:
```suggestion
  /** Reflects the `NO_COLOR` environment variable at program start.
```

**Review by lucacasonato — COMMENTED** [review]
Revert the unrelated changes please.

**Comment by sno2** [comment]
> The unrelated formatting changes should be removed from the PR.

I tried fixing them but `deno fmt` kept on doing it.  Should I still revert?

File: cli/dts/lib.deno.ns.d.ts [hunk 2]
```diff
@@ -2187,7 +2188,7 @@ declare namespace Deno {
     | HrtimePermissionDescriptor;
 
   export interface PermissionStatusEventMap {
-    "change": Event;
+    change: Event;
```
Comment (kitsonk) [inline cli/dts/lib.deno.ns.d.ts:2190]: This needs to be reverted.

**Comment by lucacasonato** [comment]
Use `./tools/format.ts` instead of `deno fmt`.

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**